packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -530,19 +537,23 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
         commandRegistry.registerCommand(CommonCommands.CLOSE_OTHER_TABS, {
             isEnabled: (event?: Event) => {
                 const tabBar = this.findTabBar(event);
-                return tabBar !== undefined && tabBar.titles.length > 1;
+                if (!tabBar) {
+                    return false;
+                }
+                const currentTitle = this.findTitle(tabBar, event);
+                return tabBar.titles.some(title => title !== currentTitle && title.closable);
```

Comment:
This condition checks only the selected tab bar, while `execute` closes all other tab bars, too.

Reply:
Ah you're right, so the original execute logic was wrong too?